apache/ode-jacob
# [WIP] Upgrade to JDK8 and latest stable Jackson release

This PR is currently work in progress and will provide the following improvements:
* [x] Update pom
  * [x] Upgrade Jackson to latest stable release
  Requires additional effort since some of the methods used from version 2.2.0-rc1 became deprecated until the latest stable version 2.9.4.
    * Change `serializeWithType(...)` implementations to comply with Jackson 2.9.4 as follows:
        ```
        @Override
        public void serializeWithType(ChannelProxy value, JsonGenerator jgen, SerializerProvider provider,
                TypeSerializer typeSer)
            throws IOException, JsonGenerationException
        {
            WritableTypeId typeId = typeSer.typeId(value, JsonToken.START_OBJECT);
            typeSer.writeTypePrefix(jgen, typeId);
            serializeContents(value, jgen, provider);
            typeSer.writeTypeSuffix(jgen, typeId);
        }
        ```
    * Change call to `ctxt.mappingException(...)` in `ContinuationDeserializer.java` and `ChannelProxyDeserializer.java` to comply with Jackson 2.9.4 as follows:
        ```
        ctxt.handleUnexpectedToken([Class], jp);
        ```
    * Serialization of @id's seems to be different, changed serialized strings in `JacksonSoupTest.java` accordingly. This needs further inspection/tests, if the JACOB implementation still behaves as expected...
  * [x] JDK 8
* [ ] Lookout for patterns in source as indicated in the CVE [ https://cve.mitre.org/cgi-bin/cvekey.cgi?keyword=jackson ].
* [ ] Test changes together with latest code from ODE master branch
